torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -100,9 +99,25 @@ def __init__(
             self.process_group = process_group
             self._per_optimizer_states = defaultdict(_refresh_per_optimizer_state)
 
+    @overload
+    def scale(self, outputs: torch.Tensor) -> torch.Tensor:
+        ...
+
+    @overload
+    def scale(self, outputs: List[torch.Tensor]) -> List[torch.Tensor]:
+        ...
+
+    @overload
+    def scale(self, outputs: Tuple[torch.Tensor, ...]) -> Tuple[torch.Tensor, ...]:
+        ...
+
+    @overload
+    def scale(self, outputs: Iterable[torch.Tensor]) -> Iterable[torch.Tensor]:
+        ...
+
     def scale(
-        self, outputs: Union[torch.Tensor, List[torch.Tensor]]
-    ) -> Union[torch.Tensor, List[torch.Tensor]]:
+        self, outputs: Union[torch.Tensor, Iterable[torch.Tensor]]
+    ) -> Union[torch.Tensor, Iterable[torch.Tensor]]:
```

Comment:
The `@overload`s here provide more specific return types than what was here before.

Reply:
```python
import torch
import torch.cuda.amp

grad_scaler = torch.cuda.amp.GradScaler()
loss = torch.tensor(1.0)

# Cannot access member "backward" for type "list[Unknown]"
# Cannot access member "backward" for type "tuple[Unknown, ...]"
# Cannot access member "backward" for type "map[Unknown]"
grad_scaler.scale(loss).backward()
```

and now:

```python
reveal_type(grad_scaler.scale(loss))  # Type of "grad_scaler.scale(loss)" is "Tensor"
reveal_type(grad_scaler.scale([loss, loss]))  # Type of "grad_scaler.scale([loss, loss])" is "List[Tensor]"
reveal_type(grad_scaler.scale((loss, loss)))  # Type of "grad_scaler.scale((loss, loss))" is "Tuple[Tensor, ...]"
reveal_type(grad_scaler.scale({loss, loss}))  # Type of "grad_scaler.scale({loss, loss})" is "Iterable[Tensor]"

```